mobly/snippet/client_base.py

```diff
@@ -127,55 +116,42 @@ def start_server(self):
       errors.ServerStartError: when failed to start the snippet server.
     """
 
-    @contextlib.contextmanager
-    def _execute_one_stage(stage):
-      """Context manager for executing one stage.
-
-      Args:
-        stage: StartServerStages, the stage which is running under this
-          context manager.
-
-      Yields:
-        None.
-      """
-      self.log.debug('[START_SERVER] Running the stage %s.', stage.name)
-      yield
-      self.log.debug('[START_SERVER] Finished the stage %s.', stage.name)
-
-    self.log.debug('Starting the server.')
+    self.log.debug('Initializing the snippet package %s.', self.package)
     start_time = time.perf_counter()
 
-    with _execute_one_stage(StartServerStages.BEFORE_STARTING_SERVER):
-      self.before_starting_server()
+    self.log.debug('Preparing to start the snippet server for %s.',
+                   self.package)
+    self.before_starting_server()
 
     try:
-      with _execute_one_stage(StartServerStages.DO_START_SERVER):
-        self.do_start_server()
-
-      with _execute_one_stage(StartServerStages.BUILD_CONNECTION):
-        self._build_connection()
+      self.log.debug('Starting the snippet server for %s.', self.package)
+      self.start_server()
 
-      with _execute_one_stage(StartServerStages.AFTER_STARTING_SERVER):
-        self.after_starting_server()
+      self.log.debug('Making a connection with the snippet server for %s.',
+                     self.package)
+      self._make_connection()
 
     except Exception:
-      self.log.error('[START SERVER] Error occurs when starting the server.')
+      self.log.error(
+          'Error occurs when initializing the snippet server for %s.',
```

Comment:
"Error occurred trying to connect to the server ..."?

Reply:
Not only the exception raised in `make_connection`, but exceptions raised in `start_server` are also caught here. 

So I modified it to "Error occurred trying to initialize the snippet server of %s."

Reply:
Modified it to "Error occurred trying to start and connect to the snippet server of %s.", which looks more precise.